build_tools/cmake/build_riscv.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Cross-compile the IREE project towards RISCV with CMake. Designed for CI,
+# but can be run manually. This uses previously cached build results and does
+# not clear build directories.
+#
+# Host binaries (e.g. compiler tools) will be built and installed in build-host/
+# RISCV binaries (e.g. tests) will be built in build-riscv/.
+
+set -x
+set -e
+
+ROOT_DIR=$(git rev-parse --show-toplevel)
+
+CMAKE_BIN=${CMAKE_BIN:-$(which cmake)}
+
+"${CMAKE_BIN?}" --version
+ninja --version
+
+cd ${ROOT_DIR?}
+
+# --------------------------------------------------------------------------- #
+# Build for the host.
+BUILD_HOST_DIR="${ROOT_DIR}/../build-host"
```

Comment:
When in a docker container this isn't necessarily going to be mounted and therefore not writeable. We mount the source root directory and a home directory. Let's put this in $HOME instead

```suggestion
BUILD_HOST_DIR="${BUILD_HOST_DIR:-$HOME/build-host}"
```

Reply:
I moved the build directory to "${BUILD_HOST_DIR:-$ROOT_DIR/build-host}". It will be following the same code path as in build_tools/cmake/build_android.sh if the variable is not set elsewhere.

Reply:
Sure that works. I put it in home for the new asan build because mixing source and build dirs is a bit icky IMO. But I think it doesn't really matter much on the CI

Reply:
I agree with you in general, which is why I put them one level up to begin with :)